src/context.ts

```diff
@@ -9,12 +10,31 @@ import type {
   TranslateFunction,
 } from './types';
 
+function wrapGenerateId<T extends Record<string, unknown>>(
```

Comment:
this is so we don't need to pass `parentId` in `http-spec`.
I'm indifferent and if this is complicated, we can just pass `parentId` explicitly.